srtcore/core.cpp

```diff
@@ -491,14 +491,14 @@ void CUDT::setOpt(SRT_SOCKOPT optName, const void* optval, int optlen)
         break;
 
     case SRTO_LINGER:
-        m_Linger = *(linger *)optval;
+        m_Linger = *(const linger *)optval;
```

Comment:
If you want that the compiler help you in anything by being const-correct, you should replace the C-style cast with `reinterpret_cast`. In C-style cast the compiler doesn't care about const at all.